galley/pkg/runtime/projections/serviceentry/converter/instance.go

```diff
@@ -83,7 +88,8 @@ func (i *Instance) convertService(service *resource.Entry, outMeta *mcp.Metadata
 	if spec.ClusterIP != "" && spec.ClusterIP != coreV1.ClusterIPNone {
 		addr = spec.ClusterIP
 	}
-	if addr == model.UnspecifiedIP && externalName == "" {
+	// Resolution NONE must not have endpoints
+	if len(endpoints) == 0 && addr == model.UnspecifiedIP && externalName == "" {
```

Comment:
why this change? Isn't `NONE` the new default if there are no endpoints?

Reply:
I think there is still chance of qualifying for `addr == model.UnspecifiedIP && externalName == ""` while having endpoints which makes NONE invalid.

Reply:
It's not obvious that's what you're doing here.  It might be better to have a cleanup condition toward the end that does something like:

```go
if resolution = networking.ServiceEntry_NONE && len(endpoints) > 0 {
   // NONE is disallowed when there are endpoints.
    resolution = networking.ServiceEntry_STATIC
}
```